pmm/pmm2-upgrade-tests.groovy

```diff
@@ -0,0 +1,180 @@
+            choices: ['2.0.0', '2.0.1'],
+            description: 'PMM Server Version to test for Upgrade',
+            name: 'DOCKER_VERSION')
+        choice(
+            choices: ['2.0.0', '2.0.1'],
+            description: 'PMM Client Version to test for Upgrade',
+            name: 'CLIENT_VERSION')
+        string(
+            defaultValue: '2.1.0',
+            description: 'dev-latest PMM Server Version',
+            name: 'PMM_SERVER_LATEST')
+    }
+    options {
+        skipDefaultCheckout()
+        disableConcurrentBuilds()
+    }
+    stages {
+        stage('Prepare') {
+            steps {
+                // clean up workspace and fetch pmm-qa repository
+                deleteDir()
+                git poll: false, branch: GIT_BRANCH, url: 'https://github.com/percona/pmm-qa.git'
+
+                slackSend channel: '#pmm-ci', color: '#FFFF00', message: "[${JOB_NAME}]: build started - ${BUILD_URL}"
+
+                sh '''
+                    sudo yum -y update --security
+                    sudo yum -y install docker
+                    sudo usermod -aG docker ec2-user
+                    sudo service docker start
+                    sudo docker stop selenoid || true && sudo docker rm selenoid || true
+                    sudo curl -s https://aerokube.com/cm/bash | bash \
+                    && sudo ./cm selenoid start --vnc --tmpfs 1000 --browsers chrome
+                    sleep 20
+                    sudo docker ps
+                    sudo docker logs selenoid
+                '''
+            }
+        }
+        stage('Start staging') {
+            steps {
+                runStaging(DOCKER_VERSION, CLIENT_VERSION, '--addclient=ps,1 --addclient=mo,2 --with-replica  --addclient=pgsql,1 --addclient=pxc,1 --with-proxysql --pmm2')
+            }
+        }
+        stage('Sanity check') {
+            steps {
+                sh 'timeout 100 bash -c \'while [[ "$(curl -s -o /dev/null -w \'\'%{http_code}\'\' \${PMM_URL}/ping)" != "200" ]]; do sleep 5; done\' || false'
+            }
+        }
+        stage('Sleep') {
+            steps {
+                sleep 60
+            }
+        }
+        stage('Check Packages before Upgrade') {
+            steps {
+                checkUpgrade(DOCKER_VERSION);
+            }
+        }
+        stage('Run Upgrade Tests & Remote Instances Tests') {
+            steps {
+                sauce('SauceLabsKey') {
+                    sauceconnect(options: '', sauceConnectPath: '') {
+                        sh """
+                            curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.34.0/install.sh | bash
+                            . ~/.nvm/nvm.sh
+                            nvm install 10.6.0
+                            sudo rm -f /usr/bin/node
+                            sudo ln -s ~/.nvm/versions/node/v10.6.0/bin/node /usr/bin/node
+                            npm install
+                            node -v
+                            npm -v
```

Comment:
do you really need it ? or it was added for debug ?

Reply:
we would need it everytime, some issue with NPM and node, hence I have added it permanently